api/src/main/java/org/pmiops/workbench/google/CloudBillingClientImpl.java

```diff
@@ -23,13 +25,32 @@ public CloudBillingClientImpl(
     this.retryHandler = retryHandler;
   }
 
-  public BillingAccount getBillingAccount(String billingAccountName) throws IOException {
+  @Override
+  public ProjectBillingInfo pollUntilBillingAccountLinked(
+      String billingAccountName, String projectId) throws IOException, InterruptedException {
+    Duration pollInterval = Duration.ofSeconds(2);
+    for (Instant deadline = Instant.now().plusSeconds(60);
+        Instant.now().isBefore(deadline);
+        Thread.sleep(pollInterval.toMillis())) {
```

Comment:
Using the for loop syntax to track the passage of time is pretty interesting, I don't think I've seen this pattern before. Unfortunately it does sleep an extra time at the end before terminating which is a bit of a waste:

```
@ 59s: polling fails
@ 59s: sleep 2
@ 61s: break loop
```

Usually the loop increment is free so this is not really a consideration. I don't feel that strongly on this though, I think this is fine if you prefer this pattern

Reply:
Thread.sleep is not the only thing that is causing clock time to pass. `getProjectBillingInfo` is a blocking call which is probably somewhat slow and inconsistent. Any time value I specify at the start of the loop below 60s is possible.

The other pattern also suffers from the issue I mentioned. This pattern seems fine to me